builds/e2e/nested-e2e.yaml

```diff
@@ -0,0 +1,38 @@
+trigger: none
+pr: none
+
+resources:
+  pipelines:
+  - pipeline: images
+    source: 'Azure-IoT-Edge-Core Build Images'
```

Comment:
You removed `trigger`, which for sure is good. You also removed `branch`, which means that unless the user explicitly gives build IDs for 'Build Images' and 'Edgelet Packages' builds, it will pick the most recent successful run _from any branch, including PRs_. I think you want to leave `branch: master` here as a default.

```
resources:
  pipelines:
  - pipeline: images
    source: 'Azure-IoT-Edge-Core Build Images'
    branch: master
  - pipeline: packages
    source: 'Azure-IoT-Edge-Core Edgelet Packages'
    branch: master
```